### PR TITLE
[25.0 backport] vendor: go.etcd.io/bbolt v1.3.10

### PR DESCRIPTION
**Pull request by austinvazquez**
**- What I did**
Backports https://github.com/moby/moby/pull/47460, https://github.com/moby/moby/pull/47933 to 25.0

**- How I did it**
```
git cherry-pick -xsS 9be820d8ca72b1c477d16700794d1b3c0f7b7515
git cherry-pick -xsS 7501b90a22391cae4ed1bc8162d3e98cb62a9438
```

**- How to verify it**
CI must be successful

**- Description for the changelog**
```markdown changelog
Vendor go.etcd.io/bbolt v1.3.10
```

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: vendor.mod [hunk 1]
```diff
@@ -4,7 +4,9 @@
 
 module github.com/docker/docker
 
-go 1.20
+go 1.21
+
+toolchain go1.22.5
```
Comment (austinvazquez) [inline vendor.mod:9]: Unintentional change when vendoring. Reverting this piece.

Comment (thaJeztah) [reply vendor.mod]: Ah, yes, ran into that myself a few times

**Comment by austinvazquez** [comment]
Well it looks like go.etcd.io/bbolt bumped their min version to Go 1.21.9 in the v1.3.10 release which would force us to drop Go 1.20 support as well.
Reference: https://github.com/etcd-io/bbolt/blob/main/CHANGELOG/CHANGELOG-1.3.md#v13102024-05-06

Go 1.20 is EOL since March 2024, so dropping support isn't the worst but as this change is a nice to have perhaps it is best to not bring it back to 25.0.

**Comment by austinvazquez** [comment]
Closing this as no plan to fix. TLDR; go.etcd.io/bbolt v1.3.10 requires Go min version bump in go.mod. Since 25.0 is maintenance, we shouldn't bump the Go min version unless required for bug/security fix.

**Pull request closed without merging**